spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,351 @@
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          component = double('component')
+          expect(described_class).to receive(:new).with(
+            tags: tags,
+            agent_base_url: agent_base_url,
+            ld_library_path: ld_library_path,
+            path_to_crashtracking_receiver_binary: path_to_crashtracking_receiver_binary,
+            logger: logger
+          ).and_return(component)
+
+          expect(component).to receive(:start)
+
+          described_class.build(settings, agent_settings, logger: logger)
+        end
+      end
+
+      context 'when missing `agent_base_url`' do
+        let(:agent_base_url) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `ld_library_path`' do
+        let(:ld_library_path) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `path_to_crashtracking_receiver_binary`' do
+        let(:path_to_crashtracking_receiver_binary) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
```

Comment:
Consider adding a test for the `warn` that gets emitted when the required parameters are `nil`.